Sources/TuistKit/Utils/ManifestGraphLoader.swift

```diff
@@ -99,32 +99,46 @@ public final class ManifestGraphLoader: ManifestGraphLoading {
         // Load Plugins
         let plugins = try await loadPlugins(at: path)
 
+        // Load Workspace
+        var allManifests = try recursiveManifestLoader.loadWorkspace(at: path)
+        let isSPMProjectOnly = allManifests.projects.isEmpty
+        let hasExternalDependencies = allManifests.projects.values.contains { $0.containsExternalDependencies }
+
         // Load DependenciesGraph
 
         let dependenciesGraph: TuistGraph.DependenciesGraph
         let packageSettings: TuistGraph.PackageSettings?
-        if let packagePath = manifestFilesLocator.locatePackageManifest(at: path) {
+
+        // Load SPM graph only if is SPM Project only or the workspace is using external dependencies
+        if let packagePath = manifestFilesLocator.locatePackageManifest(at: path),
+           isSPMProjectOnly || hasExternalDependencies
+        {
+            logger.notice("It might take a while if the cache is empty")
```

Comment:
I don't know if this line was specific for SPM @pepicrft

Reply:
You are right!